src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapSessionOptions.Linux.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace System.DirectoryServices.Protocols
+{
+    public partial class LdapSessionOptions
+    {
+        private static void PALCertFreeCRLContext(IntPtr certPtr) { /* No op */ }
+
+
+        // Options that are not supported in Linux
+
+        internal bool FQDN
+        {
+            set { /* no op */ }
```

Comment:
Open Question: For these options that are not supported on Linux is it fine to No-Op or do we usually opt for throwing?

Reply:
If it is not expected to get these called on Linux, then I would say throw Invalid Operation exception.

Reply:
I think we generally throw PNSE when an API's not supported on Linux (or whatever).